nomad/fsm.go

```diff
@@ -403,6 +403,14 @@ func (n *nomadFSM) applyAllocClientUpdate(buf []byte, index uint64) interface{}
 		return nil
 	}
 
+	// Updating the allocs with the job id and task group name
+	for _, alloc := range req.Alloc {
+		if existing, _ := n.state.AllocByID(alloc.ID); existing != nil {
+			alloc.JobID = existing.JobID
```

Comment:
If you are relying on this behavior I would add a unit test to ensure that this doesn't change